pkg/helpers/source-to-image/tar/tar.go

```diff
@@ -456,7 +457,15 @@ func (t *stiTar) extractLink(dir string, header *tar.Header, tarReader io.Reader
 }
 
 func (t *stiTar) extractFile(dir string, header *tar.Header, tarReader io.Reader) error {
-	path := filepath.Join(dir, header.Name)
+	fileName := header.Name
+	if runtime.GOOS == "windows" {
+		// ":" is special character on Windows. If file name contains ":",
+		// extraction will fail. In order to overcome this problem, we are replacing ":"
+		// to "-".
+		fileName = strings.Replace(fileName, ":", "-", -1)
```

Comment:
Would the same change make sense for non Windows cases too?

Reply:
I'd expect this to be a safe change if we enable for all hosts. But isn't `:` only special for Windows?

Reply:
It's better if there's unconditional change. Less maintenance and testing.

Reply:
There is similar conditions in rsync side https://github.com/openshift/oc/blob/8a88a4de03faf9923f41bcc06e4d68dacb4a543c/pkg/cli/rsync/pathspec.go#L32. I agree replacement without condition means less maintenantce and testing. However, In my opinion, keeping the windows condition here emphasizes the motivation of operation better, because on other hosts we don't have any reasons to perform this replacement.